contracts/TransferLimitModule.sol

```diff
@@ -0,0 +1,417 @@
+    }
+
+    function isValidThreshold(uint256 _threshold)
+        internal
+        returns (bool)
+    {
+        if  (_threshold >= 1 &&
+             _threshold < OwnerManager(manager).getThreshold()) {
+            return true;
+        }
+
+        return false;
+    }
+
+    function checkSignatures(bytes32 transactionHash, bytes signatures)
+        internal
+        view
+        returns (bool)
+    {
+        // There cannot be an owner with address 0.
+        address lastOwner = address(0);
+        address currentOwner;
+        uint256 i;
+
+        // Validate threshold is reached.
+        for (i = 0; i < threshold; i++) {
+            currentOwner = recoverKey(transactionHash, signatures, i);
+
+            // Signatures must be sorted by their address, and
+            // there shouldn't be duplicates.
+            if (currentOwner <= lastOwner) {
+                return false;
+            }
+
+            // Signer should either be one of the owners, or the delegate
+            if (currentOwner != delegate && !OwnerManager(manager).isOwner(currentOwner)) {
+                return false;
+            }
+
+            lastOwner = currentOwner;
+        }
+
+        return true;
+    }
+
+    function getEthAmount(address token, uint256 amount)
+        internal
+        view
+        returns (uint256, uint256)
+    {
+        // Amount is in wei
+        if (token == 0) {
+            return (amount, 10**18);
+        }
+
+        uint256 num;
+        uint256 den;
+        (num, den) = dutchx.getPriceOfTokenInLastAuction(token);
+        require(num != 0, "Price of token is zero");
+        require(den != 0, "Price denominator is zero");
+
+        return (SafeMath.mul(amount, num), den);
+    }
+
+    function getDaiAmount(uint256 ethNum, uint256 den)
+        internal
+        view
+        returns (uint256, uint256)
+    {
+        PriceOracleInterface priceOracle = PriceOracleInterface(dutchx.ethUSDOracle());
+        uint256 ethDaiPrice = priceOracle.getUSDETHPrice();
+        require(ethDaiPrice != 0, "USDETH price is zero");
+        return (SafeMath.mul(ethNum, ethDaiPrice), den);
+    }
+
+    function handlePayment(
+        uint256 gasUsed,
+        uint256 dataGas,
+        uint256 gasPrice,
+        address gasToken,
+        address refundReceiver
+    )
+        private
+    {
+        uint256 amount = ((gasUsed - gasleft()) + dataGas) * gasPrice;
+        // Make sure refund is within transfer limits, to prevent
+        // attacker with a compromised key to empty the safe.
+        require(handleTransferLimits(gasToken, amount), "Gas refund exceeds transfer limit");
+
+        // solium-disable-next-line security/no-tx-origin
+        address receiver = refundReceiver == address(0) ? tx.origin : refundReceiver;
+        if (gasToken == address(0)) {
+            // solium-disable-next-line security/no-send
+            require(receiver.send(amount), "Could not pay gas costs with ether");
```

Comment:
This would send tokens from the module, which is obviously not possible since there should be no funds on the module.